### PR TITLE
Make it possible to correctly indent snippet_block snippets

### PR DESCRIPTION
**Pull request by flip1995**
This adds a `indent_relative_to` arg to the `{snippet,expr}_block` functions. This makes it possible to keep the correct indentation of block like suggestions.

In addition, this makes the `trim_multiline` function private and adds a `indent_of` function, to get the indentation of the first line of a span.

The suggestion of `needless_continue` cannot be made auto applicable, since it would be also necessary to remove code following the linted expression. (Well, maybe it is possible, but I don't know how to do it. Expanding the suggestion span to the last expression, that should be removed didn't work)

changelog: Improve suggestions, when blocks of code are involved

### CODE REVIEW

File: clippy_lints/src/attrs.rs [hunk 1]
```diff
@@ -261,7 +261,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Attributes {
                                             _ => {},
                                         }
                                     }
-                                    let line_span = last_line_of_span(cx, attr.span);
+                                    let line_span = first_line_of_span(cx, attr.span);
```
Comment (flip1995) [inline clippy_lints/src/attrs.rs]: This function never returned the last line of the span, but always the first line, so I renamed it accordingly.

File: tests/ui/needless_continue.stderr [hunk 2]
```diff
@@ -47,16 +45,15 @@ LL | |             println!("Jabber");
...
-           println!("Blabber");
-           println!("Jabber");
-           ...
-
-error: This `else` block is redundant.
+   = help: consider dropping the `else` clause
+                   if (zero!(i % 2) || nonzero!(i % 5)) && i % 3 != 0 {
+                       continue;
+                   } {
+                       println!("Blabber");
```
Comment (phansch) [inline tests/ui/needless_continue.stderr]: This seems to be a regression as it now wraps the else part in a block?

Comment (flip1995) [reply tests/ui/needless_continue.stderr]: Yeah, it changed a little bit, but I would keep it that way for two reasons:
1. It's too hard to align this correctly after removing the block
2. Suggesting it like this prevents a bug, that could be produced by the suggestion:

```rust
let y: u32 = 23;
if x {
    continue;
} else {
    let y: u32 = 42;
    // ...
}

println!("{}", y); // outputs 23
```

previous, the suggestion would change the output:

```rust
let y: u32 = 23; // best case: a warning for unused var is produced
if x {
    continue;
}
let y: u32 = 42;
// ...

println!("{}", y); // outputs 42
```

now the output stays the same:

```rust
let y: u32 = 23;
if x {
    continue;
} 
{
    let y: u32 = 42;
    // ...
}

println!("{}", y); //outputs 23
```

Comment (flip1995) [reply tests/ui/needless_continue.stderr]: [Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=557334ca1cc44268d5c4649c36a4b266)

**Comment by phansch** [comment]
r=me with rebase

**Pull request merged**